fix: Use attributes for Kubernetes occurences (#1888)
codenvy/che-docs: modules/installation-guide/examples/system-variables.adoc

 `+CHE_API_INTERNAL+`,"`+http://${CHE_HOST}:${CHE_PORT}/api+`","API service internal network url. Back-end services should initiate REST communications to {prod-short} server with this URL" 
 `+CHE_WEBSOCKET_ENDPOINT+`,"`+ws://${CHE_HOST}:${CHE_PORT}/api/websocket+`","{prod-short} websocket major endpoint. Provides basic communication endpoint for major websocket interactions and messaging." 
 `+CHE_WORKSPACE_PROJECTS_STORAGE+`,"`+/projects+`","Your projects are synchronized from the {prod-short} server into the machine running each workspace. This is the directory in the machine where your projects are placed." 

>  `+CHE_WORKSPACE_PROJECTS_STORAGE_DEFAULT_SIZE+`,"`+1Gi+`","Used when {orch-name} or OpenShift-type components in a devfile request project PVC creation (Applied in case of 'unique' and 'per workspace' PVC strategy. In case of the 'common' PVC strategy, it is rewritten with the value of the `che.infra.kubernetes.pvc.quantity` property.)" 

Reviewer: ```suggestion
 `+CHE_WORKSPACE_PROJECTS_STORAGE_DEFAULT_SIZE+`,"`+1Gi+`","Used when {kubernetes} or OpenShift-type components in a devfile request project PVC creation (Applied in case of 'unique' and 'per workspace' PVC strategy. In case of the 'common' PVC strategy, it is rewritten with the value of the `che.infra.kubernetes.pvc.quantity` property.)" 
```

 In downstream will resolve as "OpenShift or OpenShift-type"

Author: Removed system-variables.adoc  from the scope of this PR.

>  `+CHE_WORKSPACE_PROJECTS_STORAGE_DEFAULT_SIZE+`,"`+1Gi+`","Used when {orch-name} or OpenShift-type components in a devfile request project PVC creation (Applied in case of 'unique' and 'per workspace' PVC strategy. In case of the 'common' PVC strategy, it is rewritten with the value of the `che.infra.kubernetes.pvc.quantity` property.)" 

Reviewer 2: I don't really understand why we keep `{orch-name} or OpenShift-type ` here ?
In the case of CRW, will it finally end up as `OpenShift or Openshift-type components...` ? That would look odd, no ?

>  `+CHE_WORKSPACE_PROJECTS_STORAGE_DEFAULT_SIZE+`,"`+1Gi+`","Used when {orch-name} or OpenShift-type components in a devfile request project PVC creation (Applied in case of 'unique' and 'per workspace' PVC strategy. In case of the 'common' PVC strategy, it is rewritten with the value of the `che.infra.kubernetes.pvc.quantity` property.)" 

Author: ```suggestion
 `+CHE_WORKSPACE_PROJECTS_STORAGE_DEFAULT_SIZE+`,"`+1Gi+`","Used when {orch-name}-type components in a devfile request project PVC creation (Applied in case of 'unique' and 'per workspace' PVC strategy. In case of the 'common' PVC strategy, it is rewritten with the value of the `che.infra.kubernetes.pvc.quantity` property.)" 
```

 `+CHE_WORKSPACE_MAVEN__OPTIONS+`,"`+-XX:MaxRAM=150m-XX:MaxRAMFraction=2 -XX:+UseParallelGC -XX:MinHeapFreeRatio=10 -XX:MaxHeapFreeRatio=20 -XX:GCTimeRatio=4 -XX:AdaptiveSizePolicyWeight=90 -Dsun.zip.disableMemoryMapping=true -Xms20m -Djava.security.egd=file:/dev/./urandom+`","Maven command-line options added to JVMs running agents in workspaces." 
 `+CHE_WORKSPACE_DEFAULT__MEMORY__LIMIT__MB+`,"`+1024+`","RAM limit default for each machine that has no RAM settings in its environment. Value less or equal to 0 is interpreted as disabling the limit." 
 `+CHE_WORKSPACE_DEFAULT__MEMORY__REQUEST__MB+`,"`+200+`","RAM request for each container that has no explicit RAM settings in its environment. This amount is allocated when the workspace container is created. This property may not be supported by all infrastructure implementations. Currently it is supported by {orch-name}. A memory request exceeding the memory limit is ignored, and only the limit size is used. Value less or equal to 0 is interpreted as disabling the limit." 

>  `+CHE_WORKSPACE_DEFAULT__CPU__LIMIT__CORES+`,"`+-1+`","CPU limit for each container that has no CPU settings in its environment. Specify either in floating point cores number, for example, `0.125`, or using the {orch-name} format, integer millicores, for example, `125m`. Value less or equal to 0 is interpreted as disabling the limit." 

Reviewer: I'm not sure it can be referred to as "OpenShift format", it is an inherently k8s thing - https://www.noqcks.io/notes/2016/12/14/kubernetes-understanding-millicores/

Author: Removed system-variables.adoc  from the scope of this PR.

 `+CHE_WORKSPACE_DEFAULT__CPU__REQUEST__CORES+`,"`+-1+`","CPU request for each container that has no CPU settings in environment. A CPU request exceeding the CPU limit is ignored, and only limit number is used. Value less or equal to 0 is interpreted as disabling the limit." 
 `+CHE_WORKSPACE_SIDECAR_DEFAULT__MEMORY__LIMIT__MB+`,"`+128+`","RAM limit and request for each sidecar that has no RAM settings in the {prod-short} plug-in configuration. Value less or equal to 0 is interpreted as disabling the limit." 
 `+CHE_WORKSPACE_SIDECAR_DEFAULT__MEMORY__REQUEST__MB+`,"`+64+`","RAMlimit and request for each sidecar that has no RAM settings in the {prod-short} plug-in configuration. Value less or equal to 0 is interpreted as disabling the limit." 

>  `+CHE_WORKSPACE_SIDECAR_DEFAULT__CPU__LIMIT__CORES+`,"`+-1+`","CPU limit and request default for each sidecar that has no CPU settings in the {prod-short} plug-in configuration. Specify either in floating point cores number, for example, `0.125`, or using the {orch-name} format, integer millicores, for example, `125m`. Value less or equal to 0 is interpreted as disabling the limit." 

Reviewer: Same as the comment above.

Author: Removed system-variables.adoc  from the scope of this PR.

>  `+CHE_WORKSPACE_SIDECAR_DEFAULT__CPU__REQUEST__CORES+`,"`+-1+`","CPUlimit and request default for each sidecar that has no CPU settings in the {prod-short} plug-in configuration. Specify either in floating point cores number, for example, `0.125`, or using the {orch-name} format, integer millicores, for example, `125m`. Value less or equal to 0 is interpreted as disabling the limit." 

Reviewer: Same as the comment above.

Btw, we'll need to think of a way to automatically apply these attributes to this file. -_-

Author: Removed system-variables.adoc  from the scope of this PR.

 `+CHE_WORKSPACE_MAVEN__OPTIONS+`,"`+-XX:MaxRAM=150m-XX:MaxRAMFraction=2 -XX:+UseParallelGC -XX:MinHeapFreeRatio=10 -XX:MaxHeapFreeRatio=20 -XX:GCTimeRatio=4 -XX:AdaptiveSizePolicyWeight=90 -Dsun.zip.disableMemoryMapping=true -Xms20m -Djava.security.egd=file:/dev/./urandom+`","Maven command-line options added to JVMs running agents in workspaces." 
 `+CHE_WORKSPACE_DEFAULT__MEMORY__LIMIT__MB+`,"`+1024+`","RAM limit default for each machine that has no RAM settings in its environment. Value less or equal to 0 is interpreted as disabling the limit." 
 `+CHE_WORKSPACE_DEFAULT__MEMORY__REQUEST__MB+`,"`+200+`","RAM request for each container that has no explicit RAM settings in its environment. This amount is allocated when the workspace container is created. This property may not be supported by all infrastructure implementations. Currently it is supported by {orch-name}. A memory request exceeding the memory limit is ignored, and only the limit size is used. Value less or equal to 0 is interpreted as disabling the limit." 

>  `+CHE_WORKSPACE_DEFAULT__CPU__LIMIT__CORES+`,"`+-1+`","CPU limit for each container that has no CPU settings in its environment. Specify either in floating point cores number, for example, `0.125`, or using the {orch-name} format, integer millicores, for example, `125m`. Value less or equal to 0 is interpreted as disabling the limit." 

Author: ```suggestion
 `+CHE_WORKSPACE_DEFAULT__CPU__LIMIT__CORES+`,"`+-1+`","CPU limit for each container that has no CPU settings in its environment. Specify either in floating point cores number, for example, `0.125`, or using the Kubernetes format, integer millicores, for example, `125m`. Value less or equal to 0 is interpreted as disabling the limit." 
```

>  `+CHE_WORKSPACE_DEFAULT__CPU__LIMIT__CORES+`,"`+-1+`","CPU limit for each container that has no CPU settings in its environment. Specify either in floating point cores number, for example, `0.125`, or using the {orch-name} format, integer millicores, for example, `125m`. Value less or equal to 0 is interpreted as disabling the limit." 

Author: ```suggestion
 `+CHE_WORKSPACE_DEFAULT__CPU__LIMIT__CORES+`,"`+-1+`","CPU limit for each container that has no CPU settings in its environment. Specify either in floating point cores number, for example, `0.125`, or using the Kubernetes format, integer millicores, for example, `125m`. Value less or equal to 0 is interpreted as disabling the limit." 
```

 `+CHE_WORKSPACE_DEFAULT__CPU__REQUEST__CORES+`,"`+-1+`","CPU request for each container that has no CPU settings in environment. A CPU request exceeding the CPU limit is ignored, and only limit number is used. Value less or equal to 0 is interpreted as disabling the limit." 
 `+CHE_WORKSPACE_SIDECAR_DEFAULT__MEMORY__LIMIT__MB+`,"`+128+`","RAM limit and request for each sidecar that has no RAM settings in the {prod-short} plug-in configuration. Value less or equal to 0 is interpreted as disabling the limit." 
 `+CHE_WORKSPACE_SIDECAR_DEFAULT__MEMORY__REQUEST__MB+`,"`+64+`","RAMlimit and request for each sidecar that has no RAM settings in the {prod-short} plug-in configuration. Value less or equal to 0 is interpreted as disabling the limit." 

>  `+CHE_WORKSPACE_SIDECAR_DEFAULT__CPU__LIMIT__CORES+`,"`+-1+`","CPU limit and request default for each sidecar that has no CPU settings in the {prod-short} plug-in configuration. Specify either in floating point cores number, for example, `0.125`, or using the {orch-name} format, integer millicores, for example, `125m`. Value less or equal to 0 is interpreted as disabling the limit." 

Author: ```suggestion
 `+CHE_WORKSPACE_SIDECAR_DEFAULT__CPU__LIMIT__CORES+`,"`+-1+`","CPU limit and request default for each sidecar that has no CPU settings in the {prod-short} plug-in configuration. Specify either in floating point cores number, for example, `0.125`, or using the Kubernetes format, integer millicores, for example, `125m`. Value less or equal to 0 is interpreted as disabling the limit." 
```

>  `+CHE_WORKSPACE_SIDECAR_DEFAULT__CPU__REQUEST__CORES+`,"`+-1+`","CPUlimit and request default for each sidecar that has no CPU settings in the {prod-short} plug-in configuration. Specify either in floating point cores number, for example, `0.125`, or using the {orch-name} format, integer millicores, for example, `125m`. Value less or equal to 0 is interpreted as disabling the limit." 

Author: ```suggestion
 `+CHE_WORKSPACE_SIDECAR_DEFAULT__CPU__REQUEST__CORES+`,"`+-1+`","CPUlimit and request default for each sidecar that has no CPU settings in the {prod-short} plug-in configuration. Specify either in floating point cores number, for example, `0.125`, or using the Kubernetes format, integer millicores, for example, `125m`. Value less or equal to 0 is interpreted as disabling the limit." 
```

 `+CHE_INFRA_KUBERNETES_NAMESPACE_LABEL+`,"`+true+`","Defines whether che-server should try to label the workspace namespaces." 
 `+CHE_INFRA_KUBERNETES_NAMESPACE_LABELS+`,"`+app.kubernetes.io/part-of=che.eclipse.org,app.kubernetes.io/component=workspaces-namespace+`","List of labels to find Namespaces/Projects that are used for {prod-short} Workspaces. They are used to:  - find prepared Namespaces/Projects for users in combination with `che.infra.kubernetes.namespace.annotations`.  - actively label namespaces with any workspace." 
 `+CHE_INFRA_KUBERNETES_NAMESPACE_ANNOTATIONS+`,"`+che.eclipse.org/username=<username>+`","List of annotations to find Namespaces/Projects prepared for {prod-short} users workspaces. Only Namespaces/Projects matching the `che.infra.kubernetes.namespace.labels` will be matched against these annotations. Namespaces/Projects that matches both `che.infra.kubernetes.namespace.labels` and `che.infra.kubernetes.namespace.annotations` will be preferentially used for User's workspaces. It's possible to use `<username>` placeholder to specify the Namespace/Project to concrete user." 

>  `+CHE_INFRA_KUBERNETES_NAMESPACE_ALLOW__USER__DEFINED+`,"`+false+`","Defines if a user is able to specify {platforms-namespace} (or OpenShift project) different from the default. It's NOT RECOMMENDED to configured true without OAuth configured. This property is also used by the OpenShift infra." 

Reviewer 2: Do we need to keep `(or OpenShift project)` here ?

>  `+CHE_INFRA_KUBERNETES_NAMESPACE_ALLOW__USER__DEFINED+`,"`+false+`","Defines if a user is able to specify {platforms-namespace} (or OpenShift project) different from the default. It's NOT RECOMMENDED to configured true without OAuth configured. This property is also used by the OpenShift infra." 

Author: ```suggestion
 `+CHE_INFRA_KUBERNETES_NAMESPACE_ALLOW__USER__DEFINED+`,"`+false+`","Defines if a user is able to specify {platforms-namespace} different from the default. It's NOT RECOMMENDED to configured true without OAuth configured. This property is also used by the OpenShift infra." 
```

 `+CHE_INFRA_KUBERNETES_PVC_ENABLED+`,"`+true+`","Defines whether use the Persistent Volume Claim for che workspace needs e.g backup projects, logs etc or disable it." 
 `+CHE_INFRA_KUBERNETES_PVC_STRATEGY+`,"`+common+`","Defined which strategy will be used while choosing PVC for workspaces. Supported strategies: - 'common'        All workspaces in the same {platforms-namespace} will reuse the same PVC.        Name of PVC may be configured with 'che.infra.kubernetes.pvc.name'.        Existing PVC will be used or new one will be created if it doesn't exist. - 'unique'        Separate PVC for each workspace's volume will be used.        Name of PVC is evaluated as '{che.infra.kubernetes.pvc.name} + '-' + {generated_8_chars}'.        Existing PVC will be used or a new one will be created if it doesn't exist. - 'per-workspace'        Separate PVC for each workspace will be used.        Name of PVC is evaluated as '{che.infra.kubernetes.pvc.name} + '-' + {WORKSPACE_ID}'.        Existing PVC will be used or a new one will be created if it doesn't exist." 

>  `+CHE_INFRA_KUBERNETES_PVC_PRECREATE__SUBPATHS+`,"`+true+`","Defines whether to run a job that creates workspace's subpath directories in persistent volume for the 'common' strategy before launching a workspace. Necessary in some versions of {orch-name} as workspace subpath volume mounts are created with root permissions, and thus cannot be modified by workspaces running as a user (presents an error importing projects into a workspace in {prod-short}). The default is 'true', but should be set to false if the version of {orch-name} creates subdirectories with user permissions. Relevant issue: \https://github.com/kubernetes/kubernetes/issues/41638 Note that this property has effect only if the 'common' PVC strategy used." 

Reviewer 2: > Necessary in some versions of {orch-name} as workspace subpath volume mounts are created with root permissions

Are we sure it is also the case on OpenShift ? cc @skabashnyuk

Reviewer 3: I believe - yes.

Author: OK, thanks.

,=== 

[id="kubernetes-infra-parameters"]

> = {orch-name} Infra parameters

Author: ```suggestion
= Kubernetes Infra parameters
```

> = {orch-name} Infra parameters

Author: ```suggestion
= Kubernetes Infra parameters
```

> .{orch-name} Infra parameters 

Author: ```suggestion
.Kubernetes Infra parameters 
```

,=== 
 Environment Variable Name,Default value, Description 
 

>  `+CHE_INFRA_KUBERNETES_MASTER__URL+`,"","Configuration of {orch-name} client that Infra will use" 

Author: ```suggestion
 `+CHE_INFRA_KUBERNETES_MASTER__URL+`,"","Configuration of Kubernetes client that Infra will use" 
```

 `+CHE_INFRA_KUBERNETES_TRUST__CERTS+`,"","Configurationof {orch-name} client that Infra will use" 
 `+CHE_INFRA_KUBERNETES_SERVER__STRATEGY+`,"`+multi-host+`","Defines the way how servers are exposed to the world in {orch-name} infra. List of strategies implemented in {prod-short}: default-host, multi-host, single-host" 

>  `+CHE_INFRA_KUBERNETES_SINGLEHOST_WORKSPACE_EXPOSURE+`,"`+native+`","Defines the way in which the workspace plugins and editors are exposed in the single-host mode. Supported exposures: - 'native': Exposes servers using {orch-name} Ingresses. Works only on {kubernetes}. - 'gateway': Exposes servers using reverse-proxy gateway." 

Author: ```suggestion
 `+CHE_INFRA_KUBERNETES_SINGLEHOST_WORKSPACE_EXPOSURE+`,"`+native+`","Defines the way in which the workspace plugins and editors are exposed in the single-host mode. Supported exposures: - 'native': Exposes servers using Kubernetes Ingresses. Works only on {kubernetes}. - 'gateway': Exposes servers using reverse-proxy gateway." 
```

 `+CHE_INFRA_KUBERNETES_SINGLEHOST_WORKSPACE_DEVFILE__ENDPOINT__EXPOSURE+`,"`+multi-host+`","Defines the way how to expose devfile endpoints, thus end-user's applications, in single-host server strategy. They can either follow the single-host strategy and be exposed on subpaths, or they can be exposed on subdomains. - 'multi-host': expose on subdomains - 'single-host': expose on subpaths" 
 `+CHE_INFRA_KUBERNETES_SINGLEHOST_GATEWAY_CONFIGMAP__LABELS+`,"`+app=che,component=che-gateway-config+`","Defines labels which will be set to ConfigMaps configuring single-host gateway." 
 `+CHE_INFRA_KUBERNETES_INGRESS_DOMAIN+`,"","Used to generate domain for a server in a workspace in case property `che.infra.kubernetes.server_strategy` is set to `multi-host`" 

>  `+CHE_INFRA_KUBERNETES_NAMESPACE+`,"","DEPRECATED - please do not change the value of this property otherwise the existing workspaces will loose data. Do not set it on new installations. Defines {platforms-namespace} in which all workspaces will be created. If not set, every workspace will be created in a new namespace, where namespace = workspace id It's possible to use <username> and <userid> placeholders (e.g.: che-workspace-<username>). In that case, new namespace will be created for each user. Service account with permission to create new namespace must be used. Ignored for OpenShift infra. Use `che.infra.openshift.project` instead If the namespace pointed to by this property exists, it will be used for all workspaces. If it does not exist, the namespace specified by the che.infra.kubernetes.namespace.default will be created and used." 

Author: ```suggestion
 `+CHE_INFRA_KUBERNETES_NAMESPACE+`,"","DEPRECATED - please do not change the value of this property otherwise the existing workspaces will loose data. Do not set it on new installations. Defines Kubernetes namespace in which all workspaces will be created. If not set, every workspace will be created in a new namespace, where namespace = workspace id It's possible to use <username> and <userid> placeholders (e.g.: che-workspace-<username>). In that case, new namespace will be created for each user. Service account with permission to create new namespace must be used. Ignored for OpenShift infra. Use `che.infra.openshift.project` instead If the namespace pointed to by this property exists, it will be used for all workspaces. If it does not exist, the namespace specified by the che.infra.kubernetes.namespace.default will be created and used." 
```

 `+CHE_INFRA_KUBERNETES_NAMESPACE_CREATION__ALLOWED+`,"`+true+`","Indicates whether {prod-short} server is allowed to create namespaces/projects for user workspaces, or they're intended to be created manually by cluster administrator. This property is also used by the OpenShift infra." 

>  `+CHE_INFRA_KUBERNETES_NAMESPACE_DEFAULT+`,"`+<username>-che+`","Defines {orch-name} default namespace in which user's workspaces are created if user does not override it. It's possible to use <username>, <userid> and <workspaceid> placeholders (e.g.: che-workspace-<username>). In that case, new namespace will be created for each user (or workspace). Is used by OpenShift infra as well to specify Project" 

Author: ```suggestion
 `+CHE_INFRA_KUBERNETES_NAMESPACE_DEFAULT+`,"`+<username>-che+`","Defines Kubernetes default namespace in which user's workspaces are created if user does not override it. It's possible to use <username>, <userid> and <workspaceid> placeholders (e.g.: che-workspace-<username>). In that case, new namespace will be created for each user (or workspace). Is used by OpenShift infra as well to specify Project" 
```

 `+CHE_INFRA_KUBERNETES_NAMESPACE_ANNOTATIONS+`,"`+che.eclipse.org/username=<username>+`","List of annotations to find Namespaces/Projects prepared for {prod-short} users workspaces. Only Namespaces/Projects matching the `che.infra.kubernetes.namespace.labels` will be matched against these annotations. Namespaces/Projects that matches both `che.infra.kubernetes.namespace.labels` and `che.infra.kubernetes.namespace.annotations` will be preferentially used for User's workspaces. It's possible to use `<username>` placeholder to specify the Namespace/Project to concrete user." 
 `+CHE_INFRA_KUBERNETES_NAMESPACE_ALLOW__USER__DEFINED+`,"`+false+`","Defines if a user is able to specify {platforms-namespace} different from the default. It's NOT RECOMMENDED to configured true without OAuth configured. This property is also used by the OpenShift infra." 

>  `+CHE_INFRA_KUBERNETES_SERVICE__ACCOUNT__NAME+`,"`+NULL+`","Defines {orch-name} Service Account name which should be specified to be bound to all workspaces pods. Note that {orch-name} Infrastructure won't create the service account and it should exist. OpenShift infrastructure will check if project is predefined(if `che.infra.openshift.project` is not empty):  - if it is predefined then service account must exist there  - if it is 'NULL' or empty string then infrastructure will create new OpenShift project per workspace    and prepare workspace service account with needed roles there" 

Author: ```suggestion
 `+CHE_INFRA_KUBERNETES_SERVICE__ACCOUNT__NAME+`,"`+NULL+`","Defines Kubernetes Service Account name which should be specified to be bound to all workspaces pods. Note that {orch-name} Infrastructure won't create the service account and it should exist. OpenShift infrastructure will check if project is predefined(if `che.infra.openshift.project` is not empty):  - if it is predefined then service account must exist there  - if it is 'NULL' or empty string then infrastructure will create new OpenShift project per workspace    and prepare workspace service account with needed roles there" 
```

 `+CHE_INFRA_KUBERNETES_WORKSPACE__START__TIMEOUT__MIN+`,"`+8+`","Defines time frame that limits the Kubernetes workspace start time" 
 `+CHE_INFRA_KUBERNETES_INGRESS__START__TIMEOUT__MIN+`,"`+5+`","Defines the timeout in minutes that limits the period for which Kubernetes Ingress become ready" 
 `+CHE_INFRA_KUBERNETES_WORKSPACE__UNRECOVERABLE__EVENTS+`,"`+FailedMount,FailedScheduling,MountVolume.SetUpfailed,Failed to pull image,FailedCreate,ReplicaSetCreateError+`","If during workspace startup an unrecoverable event defined in the property occurs, terminate workspace immediately instead of waiting until timeout Note that this SHOULD NOT include a mere 'Failed' reason, because that might catch events that are not unrecoverable. A failed container startup is handled explicitly by {prod-short} server." 
 `+CHE_INFRA_KUBERNETES_PVC_ENABLED+`,"`+true+`","Defines whether use the Persistent Volume Claim for che workspace needs e.g backup projects, logs etc or disable it." 

>  `+CHE_INFRA_KUBERNETES_PVC_STRATEGY+`,"`+common+`","Defined which strategy will be used while choosing PVC for workspaces. Supported strategies: - 'common'        All workspaces in the same {platforms-namespace} will reuse the same PVC.        Name of PVC may be configured with 'che.infra.kubernetes.pvc.name'.        Existing PVC will be used or new one will be created if it doesn't exist. - 'unique'        Separate PVC for each workspace's volume will be used.        Name of PVC is evaluated as '{che.infra.kubernetes.pvc.name} + '-' + {generated_8_chars}'.        Existing PVC will be used or a new one will be created if it doesn't exist. - 'per-workspace'        Separate PVC for each workspace will be used.        Name of PVC is evaluated as '{che.infra.kubernetes.pvc.name} + '-' + {WORKSPACE_ID}'.        Existing PVC will be used or a new one will be created if it doesn't exist." 

Author: ```suggestion
 `+CHE_INFRA_KUBERNETES_PVC_STRATEGY+`,"`+common+`","Defined which strategy will be used while choosing PVC for workspaces. Supported strategies: - 'common'        All workspaces in the same Kubernetes namespace will reuse the same PVC.        Name of PVC may be configured with 'che.infra.kubernetes.pvc.name'.        Existing PVC will be used or new one will be created if it doesn't exist. - 'unique'        Separate PVC for each workspace's volume will be used.        Name of PVC is evaluated as '{che.infra.kubernetes.pvc.name} + '-' + {generated_8_chars}'.        Existing PVC will be used or a new one will be created if it doesn't exist. - 'per-workspace'        Separate PVC for each workspace will be used.        Name of PVC is evaluated as '{che.infra.kubernetes.pvc.name} + '-' + {WORKSPACE_ID}'.        Existing PVC will be used or a new one will be created if it doesn't exist." 
```

 `+CHE_INFRA_KUBERNETES_INGRESS_PATH__TRANSFORM+`,"`+NULL+`","Defines a 'recipe' on how to declare the path of the ingress that should expose a server. The '%s' represents the base public URL of the server and is guaranteed to end with a forward slash. This property must be a valid input to the String.format() method and contain exactly one reference to '%s'. Please see the description of the che.infra.kubernetes.ingress.annotations_json property to see how these two properties interplay when specifying the ingress annotations and path. If not defined, this property defaults to '%s' (without the quotes) which means that the path is not transformed in any way for use with the ingress controller." 
 `+CHE_INFRA_KUBERNETES_INGRESS_LABELS+`,"`+NULL+`","Additional labels to add into every Ingress created by {prod-short} server to allow clear identification." 

>  `+CHE_INFRA_KUBERNETES_POD_SECURITY__CONTEXT_RUN__AS__USER+`,"`+NULL+`","Defines security context for pods that will be created by {orch-name} Infra This is ignored by OpenShift infra" 

Reviewer 2: Here we're presenting a contrast betwen Kubernetes and OpenShift infrastructure (ignored on OpenShift infra). So I assume we should use `kubernetes`

>  `+CHE_INFRA_KUBERNETES_POD_SECURITY__CONTEXT_RUN__AS__USER+`,"`+NULL+`","Defines security context for pods that will be created by {orch-name} Infra This is ignored by OpenShift infra" 

Author: ```suggestion
 `+CHE_INFRA_KUBERNETES_POD_SECURITY__CONTEXT_RUN__AS__USER+`,"`+NULL+`","Defines security context for pods that will be created by {kubernetes} Infra This is ignored by OpenShift infra" 
```

>  `+CHE_INFRA_KUBERNETES_POD_SECURITY__CONTEXT_FS__GROUP+`,"`+NULL+`","Definessecurity context for pods that will be created by {orch-name} Infra This is ignored by OpenShift infra" 

Reviewer 2: Here we're presenting a contrast betwen Kubernetes and OpenShift infrastructure (ignored on OpenShift infra). So I assume we should use `kubernetes`

>  `+CHE_INFRA_KUBERNETES_POD_SECURITY__CONTEXT_FS__GROUP+`,"`+NULL+`","Definessecurity context for pods that will be created by {orch-name} Infra This is ignored by OpenShift infra" 

Author: ```suggestion
 `+CHE_INFRA_KUBERNETES_POD_SECURITY__CONTEXT_FS__GROUP+`,"`+NULL+`","Definessecurity context for pods that will be created by {kubernetes} Infra This is ignored by OpenShift infra" 
```

 `+CHE_INFRA_KUBERNETES_INGRESS_PATH__TRANSFORM+`,"`+NULL+`","Defines a 'recipe' on how to declare the path of the ingress that should expose a server. The '%s' represents the base public URL of the server and is guaranteed to end with a forward slash. This property must be a valid input to the String.format() method and contain exactly one reference to '%s'. Please see the description of the che.infra.kubernetes.ingress.annotations_json property to see how these two properties interplay when specifying the ingress annotations and path. If not defined, this property defaults to '%s' (without the quotes) which means that the path is not transformed in any way for use with the ingress controller." 
 `+CHE_INFRA_KUBERNETES_INGRESS_LABELS+`,"`+NULL+`","Additional labels to add into every Ingress created by {prod-short} server to allow clear identification." 

>  `+CHE_INFRA_KUBERNETES_POD_SECURITY__CONTEXT_RUN__AS__USER+`,"`+NULL+`","Defines security context for pods that will be created by {kubernetes} Infra This is ignored by OpenShift infra" 

Author: ```suggestion
 `+CHE_INFRA_KUBERNETES_POD_SECURITY__CONTEXT_RUN__AS__USER+`,"`+NULL+`","Defines security context for pods that will be created by Kubernetes Infra This is ignored by OpenShift infra" 
```

>  `+CHE_INFRA_KUBERNETES_POD_SECURITY__CONTEXT_FS__GROUP+`,"`+NULL+`","Definessecurity context for pods that will be created by {kubernetes} Infra This is ignored by OpenShift infra" 

Author: ```suggestion
 `+CHE_INFRA_KUBERNETES_POD_SECURITY__CONTEXT_FS__GROUP+`,"`+NULL+`","Definessecurity context for pods that will be created by Kubernetes Infra This is ignored by OpenShift infra" 
```

 `+CHE_INFRA_KUBERNETES_POD_TERMINATION__GRACE__PERIOD__SEC+`,"`+0+`","Defines grace termination period for pods that will be created by {orch-name} infrastructures Grace termination period of {orch-name} workspace's pods defaults '0', which allows to terminate pods almost instantly and significantly decrease the time required for stopping a workspace. Note: if `terminationGracePeriodSeconds` have been explicitly set in {orch-name} recipe it will not be overridden." 
 `+CHE_INFRA_KUBERNETES_CLIENT_HTTP_ASYNC__REQUESTS_MAX+`,"`+1000+`","Number of maximum concurrent async web requests (http requests or ongoing  web socket calls) supported in the underlying shared http client of the `KubernetesClient` instances. Default values are 64, and 5 per-host, which doesn't seem correct for multi-user scenarios knowing that {prod-short} keeps a number of connections opened (e.g. for command or ws-agent logs)" 

>  `+CHE_INFRA_KUBERNETES_CLIENT_HTTP_ASYNC__REQUESTS_MAX__PER__HOST+`,"`+1000+`","Numberof maximum concurrent async web requests (http requests or ongoing  web socket calls) supported in the underlying shared http client of the `{orch-name}Client` instances. Default values are 64, and 5 per-host, which doesn't seem correct for multi-user scenarios knowing that {prod-short} keeps a number of connections opened (e.g. for command or ws-agent logs)" 

Author: ```suggestion
 `+CHE_INFRA_KUBERNETES_CLIENT_HTTP_ASYNC__REQUESTS_MAX__PER__HOST+`,"`+1000+`","Numberof maximum concurrent async web requests (http requests or ongoing  web socket calls) supported in the underlying shared http client of the `KubernetesClient` instances. Default values are 64, and 5 per-host, which doesn't seem correct for multi-user scenarios knowing that {prod-short} keeps a number of connections opened (e.g. for command or ws-agent logs)" 
```

>  `+CHE_INFRA_KUBERNETES_CLIENT_HTTP_CONNECTION__POOL_MAX__IDLE+`,"`+5+`","Max number of idle connections in the connection pool of the {orch-name}-client shared http client" 

Author: ```suggestion
 `+CHE_INFRA_KUBERNETES_CLIENT_HTTP_CONNECTION__POOL_MAX__IDLE+`,"`+5+`","Max number of idle connections in the connection pool of the Kubernetes-client shared http client" 
```

>  `+CHE_INFRA_KUBERNETES_CLIENT_HTTP_CONNECTION__POOL_KEEP__ALIVE__MIN+`,"`+5+`","Keep-alive timeout of the connection pool of the {orch-name}-client shared http client in minutes" 

Author: ```suggestion
 `+CHE_INFRA_KUBERNETES_CLIENT_HTTP_CONNECTION__POOL_KEEP__ALIVE__MIN+`,"`+5+`","Keep-alive timeout of the connection pool of the Kubernetes-client shared http client in minutes" 
```

 `+CHE_INFRA_KUBERNETES_TLS__ENABLED+`,"`+false+`","Creates Ingresses with Transport Layer Security (TLS) enabled In OpenShift infrastructure, Routes will be TLS-enabled" 
 `+CHE_INFRA_KUBERNETES_TLS__SECRET+`,"","Name of a secret that should be used when creating workspace ingresses with TLS Ignored by OpenShift infrastructure" 
 `+CHE_INFRA_KUBERNETES_TLS__KEY+`,"`+NULL+`","Data for TLS Secret that should be used for workspaces Ingresses cert and key should be encoded with Base64 algorithm These properties are ignored by OpenShift infrastructure" 
 `+CHE_INFRA_KUBERNETES_TLS__CERT+`,"`+NULL+`","Datafor TLS Secret that should be used for workspaces Ingresses cert and key should be encoded with Base64 algorithm These properties are ignored by OpenShift infrastructure" 

>  `+CHE_INFRA_KUBERNETES_RUNTIMES__CONSISTENCY__CHECK__PERIOD__MIN+`,"`+-1+`","Defines the period with which runtimes consistency checks will be performed. If runtime has inconsistent state then runtime will be stopped automatically. Value must be more than 0 or `-1`, where `-1` means that checks won't be performed at all. It is disabled by default because there is possible {prod-short} Server configuration when {prod-short} Server doesn't have an ability to interact with {orch-name} API when operation is not invoked by user. It DOES work on the following configurations: - workspaces objects are created in the same namespace where {prod-short} Server is located; - cluster-admin service account token is mount to {prod-short} Server pod; It DOES NOT work on the following configurations: - {prod-short} Server communicates with {orch-name} API using token from OAuth provider;" 

Author: ```suggestion
 `+CHE_INFRA_KUBERNETES_RUNTIMES__CONSISTENCY__CHECK__PERIOD__MIN+`,"`+-1+`","Defines the period with which runtimes consistency checks will be performed. If runtime has inconsistent state then runtime will be stopped automatically. Value must be more than 0 or `-1`, where `-1` means that checks won't be performed at all. It is disabled by default because there is possible {prod-short} Server configuration when {prod-short} Server doesn't have an ability to interact with Kubernetes API when operation is not invoked by user. It DOES work on the following configurations: - workspaces objects are created in the same namespace where {prod-short} Server is located; - cluster-admin service account token is mount to {prod-short} Server pod; It DOES NOT work on the following configurations: - {prod-short} Server communicates with Kubernetes API using token from OAuth provider;" 
```
